Sources/FuzzILTool/main.swift

```diff
@@ -16,235 +16,153 @@ import Foundation
+    print(liftToJS(program))
 }
 
-// Covert a single IL protobuf file to JS and print to stdout
-else if let ilPath = ilToJSPath {
-    var prog = Program()
-    do {
-        prog = try protobufToProgram(path: ilPath)
-    } catch {
-        print("Failed to load il proto \(ilPath) with error \(error)")
-        exit(-1)
-    }
-    let jsProgString = liftToJS(prog: prog)
-    print(jsProgString)
+// Covert a single IL protobuf file to FuzzIL's text format and print to stdout
+else if args.has("--liftToFuzzIL") {
+    let program = loadProgramOrExit(from: path)
+    print(liftToFuzzIL(program))
 }
 
 // Pretty print just the protobuf, without trying to load as a program
 // This allows the debugging of produced programs that are not syntactically valid
-else if let printPath = printProtoPath {
-    let res = try protobufToString(path: printPath)
-    print(res)
+else if args.has("--dumpProtobuf") {
+    let data = try Data(contentsOf: URL(fileURLWithPath: path))
+    let proto = try Fuzzilli_Protobuf_Program(serializedData: data)
+    dump(proto, maxDepth: 3)
 }
 
-
 // Pretty print a protobuf as a program on stdout
-else if let printPath = printProtoAsProgPath {
-    var prog = Program()
-    do {
-        prog = try protobufToProgram(path: printPath)
-    } catch {
-        print("Failed to load il proto \(printPath) with error \(error)")
-        exit(-1)
-    }
-    let prettyProgString = getPrettyStringProgram(program: prog)
-    print(prettyProgString)
+else if args.has("--dumpProgram") {
+    let program = loadProgramOrExit(from: path)
+    dump(program)
 }
 
-// Produce JS files from protobufs
-else if let dirPath = dirILToJS {
-    var isDir : ObjCBool = false
-    if !FileManager.default.fileExists(atPath: dirPath, isDirectory:&isDir) || !isDir.boolValue {
-        print("Provided directory \(dirPath) is not a valid directory path")
-        exit(-1)
-    }
-    do {
-        let numConverted = try compileAllProtosToJS(dirPath: dirPath)
-        print("Successfully converted \(numConverted) files")
-    } catch {
-        print("Failed to compile protos with error \(error)")
-        exit(-1)
-    }
+// Lift all protobuf programs to JavaScript
+else if args.has("--liftCorpusToJS") {
+    let numLiftedPrograms = liftAllPrograms(in: path, with: jsLifter, fileExtension: jsFileExtension)
+    print("Lifted \(numLiftedPrograms) programs to JS")
 }
 
-// Combine a bunch of protobufs into a state json
-else if let dirPath = combineDir {
-    var isDir : ObjCBool = false
-    if !FileManager.default.fileExists(atPath: dirPath, isDirectory:&isDir) || !isDir.boolValue {
-        print("Provided directory \(dirPath) is not a valid directory path")
-        exit(-1)
+// Combine multiple protobuf programs into a single corpus file
+else if args.has("--checkCorpus") {
+    var numPrograms = 0, numInvalidPrograms = 0
+    for (filename, program) in loadAllPrograms(in: path) {
+        numPrograms += 1
+        if !program.code.isStaticallyValid() {
```

Comment:
Would it be possible to provide additional debugging information, as to why the program is invalid?

Reply:
Yup! I implemented more verbose error reporting for program decoding. If you run --checkCorpus now with invalid programs, you'll see messages like
```
Failed to load program fuzzdir/corpus/program_20201012104930_99664821-5D15-4AF7-9EB5-A66898F96DC0.fuzzil.protobuf: decoded code is not statically valid: variable v13 was already defined
```
Does that work for you? Also let me know should you ever run into the "Failed to load program \(path) due to unexpected error" case (because that shouldn't happen)

Reply:
Great! That should greatly help in debugging